Fix issue with shift + click row range selection in the query tool data output window

KijongHan opened the pull request:
Resolves https://github.com/pgadmin-org/pgadmin4/issues/5266

# What does this PR do?
This PR resolves an apparent regression where selecting a range of rows in the query tool data output window using shift+click was no longer functional in the new pgAdmin (range selection had been recently introduced to the [react-data-grid](https://github.com/adazzle/react-data-grid) repository and the following migration of that functionality to the pgAdmin fork

# Summary of changes
- set `isShiftClick` property to `useRowSelection` react hook update function
- set `rangeSelectionMode` property to data grid components to propagate shift click behaviour to the `selectedRows` react hook under `ResultSet.jsx`

KijongHan committed:
- set rangeSelectionMode property to data grid components
- set isShiftClick property to useRowSelection react hook update function

adityatoshniwal commented on web/pgadmin/static/js/components/PgReactDataGrid.jsx:
```diff
@@ -119,7 +119,7 @@ export function CustomRow({inTest=false, ...props}) {
   return (
     <Row {...props} onKeyDown={handleKeyDown} onCellClick={onCellClick} onCellDoubleClick={onCellDoubleClick}
-      selectCell={(row, column)=>props.selectCell(row, column)} aria-selected={isRowSelected}/>
+      selectCell={(row, column)=>props.selectCell(row, column)} aria-selected={isRowSelected} rangeSelectionMode={true}/>
```
PgReactDataGrid is a common component and rangeSelectionMode true is not required here. rangeSelectionMode is only required in query tool grid.

KijongHan replied:
thanks for looking into this @adityatoshniwal - that was careless on my part. have updated so that it is only enabled from query tool component via `prop.enableRangeSelection`

adityatoshniwal commented on web/pgadmin/static/js/components/PgReactDataGrid.jsx:
```diff
@@ -158,6 +158,7 @@ export default function PgReactDataGrid({gridRef, className, hasSelectColumn=tru
           renderSortStatus,
           noRowsFallback: <Box textAlign="center" gridColumn="1/-1" p={1}>{noRowsIcon}{noRowsText || gettext('No rows found.')}</Box>,
         }}
+        enableRangeSelection={true}
```
Same as above. You should not change the common component default behaviour.

KijongHan replied:
yeah agree thanks - have updated so its set via `prop.enableRangeSelection`

adityatoshniwal commented on web/pgadmin/tools/sqleditor/static/js/components/QueryToolDataGrid/index.jsx:
```diff
@@ -116,7 +116,7 @@ function CustomRow(props) {
   return (
     <RowInfoContext.Provider value={rowInfoValue}>
-      <Row ref={rowRef} onKeyDown={handleKeyDown} {...props} />
+      <Row ref={rowRef} onKeyDown={handleKeyDown} {...props} rangeSelectionMode={true}/>
```
there is no such prop rangeSelectionMode on Row component.

KijongHan replied:
fixed, thanks

KijongHan committed:
address PR feedback
- reinstate default behaviour for range selection for components

adityatoshniwal commented on web/pgadmin/static/js/components/PgReactDataGrid.jsx:
```diff
@@ -119,7 +119,7 @@ export function CustomRow({inTest=false, ...props}) {
   return (
     <Row {...props} onKeyDown={handleKeyDown} onCellClick={onCellClick} onCellDoubleClick={onCellDoubleClick}
-      selectCell={(row, column)=>props.selectCell(row, column)} aria-selected={isRowSelected}/>
+      selectCell={(row, column)=>props.selectCell(row, column)} aria-selected={isRowSelected} rangeSelectionMode={props.enableRangeSelection}/>
```
this is not needed.

KijongHan replied:
thanks, done - I can see that `rangeSelectionMode` is mostly used as to determine the cell range selection in the underlying library, but I'm a little puzzled as to how the cell range selection is working atm without this being toggled active

adityatoshniwal replied:
that is taken care at the DataGrid main component level. subcomponents don't need the prop to be passed explicitly.

adityatoshniwal reviewed (changes requested):
While it fixes shift row selection, it doesn't work for columns and cell range selection.
I think it is a partial fix.

adityatoshniwal commented:
Hi @KijongHan,

It would be great to fix in the same PR if you can. That way we can close #5266. Otherwise, it will remain open until other things are fixed.

KijongHan committed:
- update SelectableHeaderRenderer component so it uses collection item context as opposed to having the whole collection context
- pass isShiftClick to onColumnSelected from SelectableHeaderRenderer and handle shift selection logic for columns
 #5266

KijongHan commented:
> Hi @KijongHan,
> 
> It would be great to fix in the same PR if you can. That way we can close #5266. Otherwise, it will remain open until other things are fixed.

Cool, Ill see what can be done @adityatoshniwal . I just updated the PR to introduce similar behaviour for shift+clicking through column selection if you could please review those changes

adityatoshniwal commented:
> Cool, Ill see what can be done @adityatoshniwal . I just updated the PR to introduce similar behaviour for shift+clicking through column selection if you could please review those changes

@KijongHan  Nice! So rows and columns are done. Cell range remains. Do you want to fix it?

KijongHan commented:
> > Cool, Ill see what can be done @adityatoshniwal . I just updated the PR to introduce similar behaviour for shift+clicking through column selection if you could please review those changes
> 
> @KijongHan Nice! So rows and columns are done. Cell range remains. Do you want to fix it?

Hey @adityatoshniwal - sorry didnt have too much time over the last few weeks to look into it and Im currently on holiday at the moment, if its okay with you we can merge these in minus the cell range fix?

adityatoshniwal commented:
> > > Cool, Ill see what can be done @adityatoshniwal . I just updated the PR to introduce similar behaviour for shift+clicking through column selection if you could please review those changes
> > 
> > 
> > @KijongHan Nice! So rows and columns are done. Cell range remains. Do you want to fix it?
> 
> Hey @adityatoshniwal - sorry didnt have too much time over the last few weeks to look into it and Im currently on holiday at the moment, if its okay with you we can merge these in minus the cell range fix?

OK. No problem.
